cogs/updates.py

```diff
@@ -1,38 +1,113 @@
+import logging
+from typing import Optional
+from datetime import datetime, timedelta, timezone
+
 import discord
 from discord.ext import commands
 
+from db.models import GuildData
+from utils.misc import ordinal
+from utils.checks import admin_command, staff_command, cogify
+from utils.converters import Index
+
+# updates operates by UTC dates
+UTC = timezone(offset=timedelta(), name='UTC')
+
+class UpdatesMessage(commands.Converter):
+    async def convert(self, ctx: commands.Context, message_id: str) -> discord.Message:
+        guild_data = await GuildData.get(guild_id = ctx.guild.id)
+        updates_channel = ctx.bot.get_channel(guild_data.updates_id)
+        update_message = await updates_channel.fetch_message(message_id)
+        return update_message
+
 
 class Updates(commands.Cog):
     def __init__(self, bot: commands.Bot) -> None:
         self.bot = bot
     
+    cog_check = cogify(staff_command())
+    
     @commands.hybrid_group()
     async def update(self, ctx: commands.Context):
         """
         Commands for handling updates.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        subcmds = []
+        for cmd in ctx.command.commands:
+            try:
+                if all(check(ctx) for check in cmd.checks):
+                    subcmds.append('`'+cmd.name+'`')
+            except commands.errors.MissingPermissions as e:
+                logging.debug(f'Command {cmd.name} discarded in {ctx.command.name} group subcommand list.')
+        await ctx.send(f'Available subcommands: {", ".join(subcmds)}.')
     
     @update.command()
-    async def set(self, ctx: commands.Context):
+    @admin_command()
+    async def set(self, ctx: commands.Context, channel: discord.TextChannel):
```

Comment:
The `set` command should have `channel` as an optional parameter - if `None` is provided, then the DB value should be set to None as well.